cmake_unofficial/CMakeLists.txt

```diff
@@ -1,40 +1,108 @@
-# https://cmake.org/Wiki/CMake_RPATH_handling
-SET(CMAKE_SKIP_BUILD_RPATH FALSE)
-SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+cmake_minimum_required (VERSION 2.8.6)
 
-# Where we search for shared libraries
-SET(CMAKE_INSTALL_RPATH "./lib")
+# If XXHASH is being bundled in another project, we don't want to
+# install anything.  However, we want to let people override this, so
+# we'll use the XXHASH_BUNDLED_MODE variable to let them do that; just
+# set it to OFF in your project before you add_subdirectory(xxhash/contrib/cmake_unofficial).
+get_directory_property(XXHASH_PARENT_DIRECTORY PARENT_DIRECTORY)
+if("${XXHASH_BUNDLED_MODE}" STREQUAL "")
+  # Bundled mode hasn't been set one way or the other, set the default
+  # depending on whether or not we are the top-level project.
+  if("${XXHASH_PARENT_DIRECTORY}" STREQUAL "")
+    set(XXHASH_BUNDLED_MODE OFF)
+  else()
+    set(XXHASH_BUNDLED_MODE ON)
+  endif()
+endif()
+mark_as_advanced(XXHASH_BUNDLED_MODE)
 
-# add the automatically determined parts of the RPATH
-# which point to directories outside the build tree to the install RPATH
-SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+# Allow people to choose whether to build shared or static libraries
+# via the BUILD_SHARED_LIBS option unless we are in bundled mode, in
+# which case we always use static libraries.
+include(CMakeDependentOption)
+CMAKE_DEPENDENT_OPTION(BUILD_SHARED_LIBS "Build shared libraries" ON "NOT XXHASH_BUNDLED_MODE" OFF)
 
-add_library(xxhash ../xxhash.c)
-set_target_properties(xxhash PROPERTIES COMPILE_DEFINITIONS "XXHASH_EXPORT"
-                       VERSION "${XXHASH_LIB_VERSION}"
-                       SOVERSION "${XXHASH_LIB_SOVERSION}")
+include_directories("${XXHASH_TOP_SOURCE_DIR}")
 
-if (BUILD_XXHSUM)
-    add_executable(xxhsum ../xxhsum.c)
-    target_link_libraries(xxhsum xxhash)
+# Whether to use position independent code for the static library.  If
+# we're building a shared library this is ignored and PIC is always
+# used.
+option(XXHASH_POSITION_INDEPENDENT_LIB "Use position independent code for static library (if applicable)" ON)
+if(XXHASH_POSITION_INDEPENDENT_LIB OR BUILD_SHARED_LIBS)
+  set(XXHASH_POSITION_INDEPENDENT_CODE TRUE)
+else()
+  set(XXHASH_POSITION_INDEPENDENT_CODE FALSE)
 endif()
 
-INSTALL(FILES ../xxhash.h DESTINATION include)
-INSTALL(
-    TARGETS xxhash xxhsum
-    RUNTIME DESTINATION bin
-    ARCHIVE DESTINATION lib
-    LIBRARY DESTINATION lib
-)
+# libxxhash
+add_library(xxhash "${XXHASH_TOP_SOURCE_DIR}/xxhash.c")
+set_target_properties(xxhash PROPERTIES
+  SOVERSION "${XXHASH_VERSION_STRING}"
+  VERSION "${XXHASH_VERSION_STRING}"
+  POSITION_INDEPENDENT_CODE ${XXHASH_POSITION_INDEPENDENT_CODE})
+
+# xxhsum
+add_executable(xxhsum "${XXHASH_TOP_SOURCE_DIR}/xxhsum.c")
+target_link_libraries(xxhsum xxhash)
+
+# Extra warning flags
+include (CheckCCompilerFlag)
+foreach (flag
+    -Wall -Wextra -Wcast-qual -Wcast-align -Wshadow
```

Comment:
LOL.  I stole these from [the Makefile](https://github.com/Cyan4973/xxHash/blob/dev/Makefile#L34), I don't have a strong opinion on what to put in there, that's up to Yann.  There are a bunch of [possibilities in Squash](https://github.com/quixdb/squash/blob/master/cmake/ExtraWarningFlags.cmake), though.  -Wpedantic isn't one of them (it's quite irritating, and IMHO not worth the noise).